### ISSUE TITLE
THREE.js

### ISSUE THREAD

**Issue by HMUDesign** [mikeseven/node-webgl#34]
I finally got node-webgl to compile and run. I am now trying to get it to threejs to run.

I firstly had to modify threejs to add "#ifdef GL_ES ... #endif" around the precision definitions, but that wasn't difficult.

Now it is running, opening the gl window, then exiting without errors. You can get my code at https://gist.github.com/HMUDesign/91771713413422db85be

I have narrowed down the problem a bit. It seems some of the calls are causing a crash. In the for loop on line 22385, inside loadUniformsGeneric, Certain of these uniforms simply cause the application to exit.

Any help or thoughts you may have would be greatly appreciated. As a note, all of the examples run fine.

**Comment by yapus** [comment]
+1. Also trying to run THREE.WebGLRenderer inside node-webgl

It seems for me that these console messages:
<pre>
Status: Using GLEW 1.11.0
THREE.WebGLRenderer 53
THREE.WebGLRenderer: Float textures not supported.
THREE.WebGLRenderer: Standard derivatives not supported.
THREE.WebGLRenderer: S3TC compressed textures not supported.

TypeError: Expected bindTexture(number target, WebGLTexture texture)
    at Object.bindTexture (~/work/src/node_modules/node-webgl/lib/webgl.js:81:11)
    at setTexture (eval at <anonymous> (~/work/src/node_modules/three.js/lib/three.js:31:3), <anonymous>:22088:8)
    at loadUniformsGeneric (eval at <anonymous> (~/work/src//node_modules/three.js/lib/three.js:31:3), <anonymous>:21059:12)
</pre>

have something to do with the specific scene (3d earth globe  by Bjorn Sandvik - thematicmapping.org) and the textures used. Any suggestions?

**Comment by mikeseven** [comment]
Now that node4/nan2 support is in, let's revisit three.js support. Can u try and make sure you use latest glfw and glew and node bindings too.

**Comment by HMUDesign** [comment]
I had to reinstall my whole build environment. Probably something related to not having done anything with it since v0.10...

Anyway. I also had to manually install node-glfw. I will submit an issue there with the reason for it.

Once everything was installed, threejs worked fine after a few tweaks.
* It makes references to "self" which had to be adjusted. It's polyfills dont really work in nodejs, as expected.
* I had to add `#ifdef GL_ES` several times, and once in a weird way.
* threejs makes references to getShaderPrecisionFormat to determine which precision to use, but as this version doesn't support htat at all, the function does not exist.

After the above, works great. The requestAnimationFrame provided is currently rendering the threejs quick-start cube at 945 fps...
